misk/src/main/kotlin/misk/clustering/DefaultCluster.kt

```diff
@@ -26,7 +26,7 @@ internal class DefaultCluster(
       resourceMapper = newResourceMapper(setOf())
   ))
   private val running = AtomicBoolean(false)
-  private val actions = BlockingArrayQueue<(MutableSet<ClusterWatch>) -> Unit>()
+  private val actions = LinkedBlockingQueue<(MutableSet<ClusterWatch>) -> Unit>()
```

Comment:
`BlockingArrayQueue` is effectively unbounded and grows up to `Integer.MAX_VALUE`. `LinkedBlockingQueue` defaults to allow `Integer.MAX_VALUE`, behaviour should be the same. 

This is what `Executors.newSingleThreadExecutor` uses by default